src/api/app/controllers/webui/distributions_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Webui::DistributionsController < Webui::WebuiController
+  before_action :set_project
+  before_action :set_distribution, only: [:toggle]
+  after_action :verify_authorized
+
+  # GET /projects/:project/distributions/new
+  def new
+    authorize @project, :update?
+
+    @distributions = Distribution.all.group_by(&:vendor)
+    return if @distributions.present?
+
+    if User.admin_session?
+      redirect_to(new_interconnect_path,
+                  alert: 'There are no distributions configured. Maybe you want to connect to one of the public OBS instances?')
+    else
+      redirect_to(project_repositories_path(project: @project))
+    end
+  end
+
+  # PATCH /projects/:project_name/distributions/:id/toggle
+  def toggle
+    authorize @project, :update?
+
+    @repository = @project.repositories.find_by(name: @distribution.reponame)
+    if @project.has_distribution(@distribution.project, @distribution.repository)
+      destroy_repository
+    else
+      create_repository_from_distribution
+    end
+  end
+
+  private
+
+  def create_repository_from_distribution
+    repository = Repository.new_from_distribution(@distribution)
+    repository.project = @project
+
+    if repository.save
+      @project.store(comment: "Added #{repository.name} repository")
+    else
+      flash.now[:error] = "Can not add repository: #{repository.errors.full_messages.to_sentence}"
+    end
+  end
+
+  def destroy_repository
+    @project.repositories.delete(@repository)
+    if @project.valid?
+      @project.store(comment: "Removed #{@repository.name} repository")
+    else
+      flash.now[:error] = "Failed to remove repository: #{@project.errors.full_messages.to_sentence}"
+    end
+  end
+
+  def set_distribution
+    @distribution = Distribution.find(params[:distribution])
```

Comment:
I don't see the point of having a method which is then a `before` callback if it's only used in one action. It can be done in the action directly and simplify the code.